src/main/java/chess/chessgame/Chessboard.java

```diff
@@ -0,0 +1,138 @@
+package chess.chessgame;
+
+import chess.piece.*;
+import chess.utils.ChessboardGenerator;
+
+import java.util.*;
+
+public class Chessboard {
+
+    private final Map<Position, Piece> board;
+
+    public Chessboard(ChessboardGenerator chessboardGenerator) {
+        board = chessboardGenerator.generate();
+    }
+
+    public void move(MovingPosition movingPosition, Turn turn) {
+        validate(movingPosition, turn);
+        movePiece(movingPosition);
+    }
+
+    public boolean isOver() {
+        return (int) board.values()
+                .stream()
+                .filter(piece -> piece.isSameType(Type.KING))
+                .count() == 1;
+    }
+
+    public double computeScore(Color color, double minusScoreOfSameYPawn) {
+        double score = computeTotalScore(color);
+        score += computeMinusScore(minusScoreOfSameYPawn);
+        return score;
+    }
+
+    public Map<Position, Piece> getBoard() {
+        return Collections.unmodifiableMap(board);
+    }
+
+    private void validate(MovingPosition movingPosition, Turn turn) {
+        Piece from = board.get(movingPosition.getFrom());
+        Piece to = board.get(movingPosition.getTo());
+
+        validateBlank(from);
+        validateTurn(from, turn);
+        validateColor(from, to);
+        validateMovable(movingPosition, from, to);
+        validateMiddlePosition(movingPosition, from);
+    }
+
+    private void validateBlank(Piece piece) {
+        if (piece.isSameType(Type.BLANK)) {
```

Comment:
```suggestion
        if (piece.isBlank()) {
```

isSameType이 쓰이는 방법이 blank와 pawn 밖에 없으니 메서드를 추가해도 괜찮지 않을까요

Reply:
isSameType이 쓰이는 곳은 blank,pawn,king 세군데 입니다! king은 전체 체스판에서 king의 개수를 계산하여 게임종료여부를 판단하는 Chessboard.isOver()에서 사용되고 있습니다.  두군데라면 메서드를 추가하는 것이 좋다고 느껴집니다만, 세군데이며 더 늘어날지도 모르는 상황을 대비하여 해당 메서드를 사용하는 것이 어떨까 라는 생각이 듭니다. 이에 대한 구구의 의견을 여쭤봐도 될까요?

Reply:
상황에 따라 다르겠네요. 아리가 결정해서 적용하시면 됩니다.

Reply:
넵 감사합니다! :)